### PR TITLE
Use new,  struct-based `ex_dtls`; add tests

### PR DESCRIPTION
**Pull request by LVala**

### CODE REVIEW

File: lib/ex_webrtc/dtls_transport.ex [hunk 1]
```diff
@@ -42,19 +42,18 @@ defmodule ExWebRTC.DTLSTransport do
...
-    {:ok, pkey} = ExDTLS.get_pkey(cert_client)
-    {:ok, fingerprint} = ExDTLS.get_cert_fingerprint(cert_client)
-    :ok = ExDTLS.stop(cert_client)
+    dtls = ExDTLS.init(client_mode: true, dtls_srtp: true)
+    cert = ExDTLS.get_cert(dtls)
+    pkey = ExDTLS.get_pkey(dtls)
+    fingerprint = ExDTLS.get_cert_fingerprint(dtls)
 
-    {:ok, ice_agent} = ICEAgent.start_link(:controlled, ice_config)
+    {:ok, ice_agent} = ice_module.start_link(:controlled, ice_config)
```
Comment (mickel8) [inline lib/ex_webrtc/dtls_transport.ex:52]: To be perfect we should define a behavior for `ice_module`

File: lib/ex_webrtc/dtls_transport.ex [hunk 2]
```diff
@@ -118,8 +117,25 @@ defmodule ExWebRTC.DTLSTransport do
   end
 
   @impl true
-  def handle_info({:ex_dtls, _from, msg}, state) do
-    state = handle_dtls(msg, state)
+  def handle_info(
+        :dtls_timeout,
+        %{ice_state: ice_state, buffered_packets: buffered_packets} = state
+      ) do
+    case ExDTLS.handle_timeout(state.dtls) do
```
Comment (mickel8) [inline lib/ex_webrtc/dtls_transport.ex:124]: what a brilliant ExDTLS api, can't believe

File: test/ex_webrtc/dtls_transport_test.exs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  end
+
+  setup do
+    assert {:ok, dtls} = DTLSTransport.start_link([tester: self()], FakeICEAgent)
+
+    %{dtls: dtls}
+  end
+
+  test "forwards non-data ICE messages", %{dtls: dtls} do
+    message = "test message"
```
Comment (mickel8) [inline test/ex_webrtc/dtls_transport_test.exs:34]: better send some signaling message like connection state change albo gathering state change

File: test/ex_webrtc/dtls_transport_test.exs [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    refute_receive({:fake_ice, _msg})
+  end
+
+  test "will retransmit after initiating handshake", %{dtls: dtls} do
+    :ok = DTLSTransport.start_dtls(dtls, :active)
+
+    send_ice(dtls, :connected)
+
+    assert_receive {:fake_ice, _packets}
+    assert_receive {:fake_ice, _retransmited}, 1200
```
Comment (mickel8) [inline test/ex_webrtc/dtls_transport_test.exs:77]: assert packets and retransmitted are the same

File: test/ex_webrtc/dtls_transport_test.exs [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+    send_ice(dtls, {:data, packets})
+    refute_receive {:fake_ice, _packets}
+
+    send_ice(dtls, :connected)
+    assert_receive {:fake_ice, packets}
+    assert is_binary(packets)
+  end
+
+  test "finishes handshake in actice mode", %{dtls: dtls} do
```
Comment (mickel8) [inline test/ex_webrtc/dtls_transport_test.exs:94]:
```suggestion
  test "finishes handshake in active mode", %{dtls: dtls} do
```

File: test/ex_webrtc/dtls_transport_test.exs [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+      {:handshake_finished, _, _, _, packets} ->
+        send_ice(dtls, {:data, packets})
+        :ok
+
+      {:handshake_finished, _, _, _} ->
+        :ok
+    end
+  end
+
+  defp send_ice(dtls, msg), do: send(dtls, {:ex_ice, "dummy_pid", msg})
```
Comment (mickel8) [inline test/ex_webrtc/dtls_transport_test.exs:131]: I would make this accept ice_pid instead i.e

```suggestion
  defp send_ice(ice_agent, msg), do: send(ice_agent, {:send, msg})
```

**Review by mickel8 — APPROVED** [review]

File: lib/ex_webrtc/dtls_transport.ex (hunk 1 above)
Comment (LVala) [reply lib/ex_webrtc/dtls_transport.ex:52]: Maybe let's let it be for now, especially considering that `DTLSTransport` uses exactly one function from `ICEAgent` API.

File: test/ex_webrtc/dtls_transport_test.exs (hunk 4 above)
Comment (LVala) [reply test/ex_webrtc/dtls_transport_test.exs:77]: But they are not the same tho.

**Pull request merged**